zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -120,29 +131,179 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
         } else if ("endDate".equals(field)) {
           builder.endDate();
         } else {
-          throw new RuntimeException("unkown sortBy " + field);
+          // Collect validation error for unknown sortBy field
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field));
+          continue;
```

Comment:
❓ Why do we move to the next sorting entry here already? We could simply validate the whole entry and collect all possible validation errors.